bodhi/server/tasks/__init__.py

```diff
@@ -0,0 +1,85 @@
+# Copyright © 2018-2019 Red Hat, Inc. and others.
+#
+# This file is part of Bodhi.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""Asynchronous tasks for Bodhi."""
+
+import logging
+import sys
+
+from celery import Celery
+
+from bodhi.server import bugs, buildsys, initialize_db
+from bodhi.server.config import config
+from bodhi.server.util import pyfile_to_module
+
+
+# Workaround https://github.com/celery/celery/issues/5416
+if sys.version_info >= (3, 7):
+    from re import Pattern
+    from celery.app.routes import re as routes_re
+    routes_re._pattern_type = Pattern
+
+log = logging.getLogger('bodhi')
+
+# The Celery app object.
+app = Celery()
+app.config_from_object(pyfile_to_module(config["celery_config"], "celeryconfig"))
```

Comment:
That won't work because `app` is used as a decorator before the tasks are loaded, so it needs to be configured first.

I think the issue is that the development.ini file that is used in the vagrant box is copied from the file in `devel/`, so it needs to be updated with the changes that happened in this PR (unless you start with a fresh vagrant box on this branch)

Reply:
Hmm. `Celery.config_from_object` lets users specify module names as strings, which only tries to import the module when it's used for the first time. Let me cook something up using a PEP 302 custom finder loader and see if it helps.

Reply:
As discussed, this just needs the appropriate configuration in `development.ini` (cf. `devel/development.ini.example`).

Reply:
The `development.ini` file in Vagrant is created by copying the `devel/development.ini.example` file at provision time. Are you referring ton another file?